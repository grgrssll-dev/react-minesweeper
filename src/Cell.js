import React, { PureComponent } from 'react';
import './Cell.css';

class Cell extends PureComponent {

	_getValue(number, isFlagged) {
		if (isFlagged) {
			return (<span role="img" aria-label="flag">ðŸš©</span>);
		} else if (number === -1) {
			return (<span role="img" aria-label="mine">ðŸ’£</span>);
		} else if (number < 9 && number > 0) {
			return number;
		} else {
			return ' ';
		}
	}

	_isRightClick(e) {
		return (e.type === 'click' && (e.button === 2 || e.which === 3));
	}

	render() {
		const { triggered, x, y, number, isFlagged, isRevealed, onCellClick } = this.props;
		const triggeredClass = triggered ? 'cell-triggered' : ''
		console.log('-- cell render', x, y);
		return (
			<div className={`cell ${triggeredClass}`}
				data-number={number}
				onClick={(e) => {
					onCellClick(this._isRightClick(e), x, y);
					e.preventDefault();
				}}
				onContextMenu={(e) => {
					onCellClick(true, x, y);
					e.preventDefault();
				}}>
				<div className="cell--inner"
					data-revealed={isRevealed}
					data-flagged={isFlagged}>
					{this._getValue(number, isFlagged)}
				</div>
			</div>
		);
	}
}

export default Cell;
