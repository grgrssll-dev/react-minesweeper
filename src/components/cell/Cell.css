.cell {
	box-sizing: border-box;
	border: 0.0625rem var(--grid-color) solid;
	border-top: 0;
	border-left: 0;
	cursor: default;
	display: inline-block;
	font-size: 1rem;
	height: 1rem;
	width: 1rem;
	vertical-align: top;
}

.cell--inner {
	border-bottom-color: var(--game-border-dark);
	border-left-color: var(--game-border-light);
	border-right-color: var(--game-border-dark);
	border-style: solid;
	border-top-color: var(--game-border-light);
	border-width: 0.125rem;
	box-sizing: border-box;
	color: transparent;
	cursor: default;
	font-family: 'VT323', monospace;
	font-size: 1.25rem;
	height: calc(1rem - 0.0625rem);
	line-height: 0.8;
	text-align: center;
	width: calc(1rem - 0.0625rem);
}

.cell--inner[data-revealed="true"][data-flagged="false"] {
	border-width: 0;
}

.cell[data-number="0"] [data-revealed="true"] {
	color: rgba(0, 0, 0, 0.2);
}

.cell[data-number="1"] [data-revealed="true"] {
	color: var(--val-1);
}

.cell[data-number="2"] [data-revealed="true"] {
	color: var(--val-2);
}

.cell[data-number="3"] [data-revealed="true"] {
	color: var(--val-3);
}

.cell[data-number="4"] [data-revealed="true"] {
	color: var(--val-4);
}

.cell[data-number="5"] [data-revealed="true"] {
	color: var(--val-5);
}

.cell[data-number="6"] [data-revealed="true"] {
	color: var(--val-6);
}

.cell[data-number="7"] [data-revealed="true"] {
	color: var(--val-7);
}

.cell[data-number="8"] [data-revealed="true"] {
	color: var(--val-8);
}

.cell [data-flagged="true"] {
	color: var(--system-text);
	font-size: 0.5rem;
	line-height: 1.6;
}

[data-game-over="true"] .cell[data-number="-1"] .cell--inner {
	color: var(--system-text);
	font-size: 0.6rem;
	line-height: 1.6;
}

.cell.cell-triggered .cell--inner {
	border: 0;
	background: var(--error);
}