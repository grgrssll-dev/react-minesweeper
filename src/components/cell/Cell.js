import React from 'react';
import './Cell.css';

function cell(props) {
	const {
		triggered,
		x,
		y,
		number,
		isFlagged,
		isRevealed,
		onCellClick
	} = props;

	const getValue = (number, isFlagged) => {
		if (isFlagged) {
			return (<span role="img" aria-label="flag">ðŸš©</span>);
		} else if (number === -1) {
			return (<span role="img" aria-label="mine">ðŸ’£</span>);
		} else if (number < 9 && number > 0) {
			return number;
		} else {
			return ' ';
		}
	}

	const isRightClick = (e) => {
		return (e.type === 'click' && (e.button === 2 || e.which === 3));
	}

	console.log('-- cell render', y, x);
	return (
		<div className={`cell ${triggered ? 'cell-triggered' : ''}`}
			data-number={number}
			onClick={(e) => {
				onCellClick(isRightClick(e), x, y);
				e.preventDefault();
			}}
			onContextMenu={(e) => {
				onCellClick(true, x, y);
				e.preventDefault();
			}}>
			<div className="cell--inner"
				data-revealed={isRevealed}
				data-flagged={isFlagged}>
				{getValue(number, isFlagged)}
			</div>
		</div>
	);
}

export default cell;
