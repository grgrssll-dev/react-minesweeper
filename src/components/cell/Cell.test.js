import React from 'react';
import { shallow } from 'enzyme/build';
import Cell from './Cell';

const baseProps = {
	triggered: false,
	x: 0,
	y: 0,
	number: 3,
	isFlagged: false,
	isRevealed: false,
	onCellClick: jest.fn(),
	isGameOver: false,
	className: 'class-name',
};

describe('Cell [Unit]', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	it('should match component snapshot', () => {
		expect.assertions(1);
		const props = {
			...baseProps,
		};
		const wrapper = shallow(<Cell {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});