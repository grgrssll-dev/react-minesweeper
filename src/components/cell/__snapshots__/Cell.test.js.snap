// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell [Unit] should match component snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
